### PR TITLE
Rename ironic-rhcos to ironic-machine-os

### PR DESCRIPTION
**Pull request by elfosardo**
Also temporary removing 4.5 template until 4.5 base image is made
available.

### CODE REVIEW

**Comment by sosiouxme** [comment]
Will need a corresponding ocp-build-data change for downstream, file a PR / ticket with ART

**Comment by elfosardo** [comment]
@sosiouxme  PRs and ticket created
I'm afraid though this will never pass until the CI is stable

**Comment by sosiouxme** [comment]
@elfosardo looks like there are conflicts to solve now. I think the CI is on better footing now?
/retest
going to back out the matching 4.2 downstream change until this gets in.

**Comment by elfosardo** [comment]
@sosiouxme  please review at your earliest convenience, thanks!

**Comment by sosiouxme** [comment]
@elfosardo of course now it has conflicts.
I think I can give a /lgtm however I'm not an approver -- looks like only @smarterclayton is.

File: ci-operator/config/openshift/ironic-rhcos-downloader/openshift-ironic-rhcos-downloader-release-4.5.yaml [hunk 1]
```diff
@@ -1,27 +0,0 @@
-base_images:
```
Comment (smarterclayton) [inline ci-operator/config/openshift/ironic-rhcos-downloader/openshift-ironic-rhcos-downloader-release-4.5.yaml]: did new versions of these files get created somewhere?

Comment (elfosardo) [reply ci-operator/config/openshift/ironic-rhcos-downloader/openshift-ironic-rhcos-downloader-release-4.5.yaml]: yes, but I don't know at which point, I haven't double-checked
the problem is that the 4.5 base image doesn't exist and it's blocking the CI
once this is merged I will add the files back and wait for it to be available, but at the moment I don't think we need it

**Comment by sosiouxme** [comment]
@elfosardo needs a rebase. We should really, really try to have this done in advance of code freeze...

**Pull request merged**